schema-registry/authorizer/authorization-agent/src/main/java/com/hortonworks/registries/schemaregistry/authorizer/agent/DefaultAuthorizationAgent.java

```diff
@@ -0,0 +1,515 @@
+
+            return new Authorizer.SchemaVersionResource(sGroup, sName, sBranch);
+        });
+    }
+
+    @Override
+    public void authorizeDeleteSchemaMetadata(UserAndGroups userAndGroups,
+                                              ISchemaRegistry schemaRegistry,
+                                              String schemaMetadataName)
+            throws AuthorizationException, SchemaNotFoundException {
+
+        SchemaMetadataInfo smi = schemaRegistry.getSchemaMetadataInfo(schemaMetadataName);
+        if(smi == null) {
+            throw new SchemaNotFoundException("No SchemaMetadata exists with key: " + schemaMetadataName);
+        }
+
+        Authorizer.SchemaMetadataResource schemaMetadataResource = new Authorizer.SchemaMetadataResource(
+                smi.getSchemaMetadata().getSchemaGroup(), schemaMetadataName);
+        authorize(schemaMetadataResource, AccessType.DELETE, userAndGroups);
+
+        Collection<SchemaBranch> branches = schemaRegistry.getSchemaBranches(schemaMetadataName);
+        if(branches != null) {
+            for(SchemaBranch branch : branches) {
+                authorizeDeleteSchemaBranch(userAndGroups, schemaRegistry, branch.getId());
+            }
+        }
+    }
+
+    @Override
+    public void authorizeSchemaMetadata(UserAndGroups userAndGroups,
+                                        ISchemaRegistry schemaRegistry,
+                                        String schemaMetadataName,
+                                        AccessType accessType)
+            throws AuthorizationException, SchemaNotFoundException {
+
+        SchemaMetadataInfo smi = schemaRegistry.getSchemaMetadataInfo(schemaMetadataName);
+        if(smi == null) {
+            throw new SchemaNotFoundException("No SchemaMetadata exists with key: " + schemaMetadataName);
+        }
+        authorizeSchemaMetadata(userAndGroups,
+                smi,
+                accessType);
+    }
+
+    @Override
+    public void authorizeSchemaMetadata(UserAndGroups userAndGroups,
+                                        SchemaMetadataInfo schemaMetadataInfo,
+                                        AccessType accessType)
+            throws AuthorizationException {
+
+        authorizeSchemaMetadata(userAndGroups,
+                schemaMetadataInfo.getSchemaMetadata(),
+                accessType);
+    }
+
+
+    @Override
+    public void authorizeSchemaMetadata(UserAndGroups userAndGroups, SchemaMetadata schemaMetadata,
+                                         AccessType accessType) throws AuthorizationException {
+
+        String sGroup = schemaMetadata.getSchemaGroup();
+        String sName = schemaMetadata.getName();
+
+        Authorizer.SchemaMetadataResource schemaMetadataResource = new Authorizer.SchemaMetadataResource(sGroup, sName);
+        authorize(schemaMetadataResource, accessType, userAndGroups);
+
+        if(accessType == AccessType.DELETE) {
```

Comment:
We allow schema metadata deletion :)

Reply:
Yes, we allow, but I use a different method to perform the check. authorizeDeleteSchemaMetadata() method is used.